res/controllers/Hercules-DJControl-Inpulse-300-script.js

```diff
@@ -44,76 +60,233 @@ DJCi300.scratchShiftMultiplier = 4;
+        2: false
+    };
+
+    // Pad mode variables
+    // Initialize to padModeNone
+    DJCi300.padMode = {
+        1: DJCi300.padModeNone,
+        2: DJCi300.padModeNone
+    };
+
+    // Toneplay variables
+    // Toneplay offset (shifts the toneplay keyboard up or down)
+    DJCi300.toneplayOffset = {
+        1: 0,
+        2: 0
+    };
+
+    // Slicer variables
+    // Slicer storage (stores slicer button positions)
+    DJCi300.slicerPoints = {
+        1: [-1, -1, -1, -1, -1, -1, -1, -1, -1],
+        2: [-1, -1, -1, -1, -1, -1, -1, -1, -1]
+    };
+    // Slicer buttons (stores whether or not slicer button is pressed)
+    DJCi300.slicerButtonEnabled = {
+        1: [false, false, false, false, false, false, false, false],
+        2: [false, false, false, false, false, false, false, false]
+    };
+    // Slicer beat (stores what beat slicer is on)
+    DJCi300.slicerBeat = {
+        1: 0,
+        2: 0
+    };
+
+    // Slicer connections
+    DJCi300.slicerConnections = {
+        size: [undefined, undefined], // Connected to beatloop_size
+        beat: [undefined, undefined], // Connected to beat_distance
+        load: [undefined, undefined], // Connected to LoadSelectedTrack
+    }
+
+    // Loop variables
+    // loopMode is on (equal to 1) if the current loop is created by the "Loop In" button
+    // and off (equal to 0) if it is created by the slicer buttons
+    DJCi300.loopMode = {
+        1: 0,
+        2: 0
     };
 
     // Turn On Vinyl buttons LED(one for each deck).
     midi.sendShortMsg(0x91, 0x03, 0x7F);
     midi.sendShortMsg(0x92, 0x03, 0x7F);
 
-    //Turn On Browser button LED
+    // Turn On Browser button LED
     midi.sendShortMsg(0x90, 0x04, 0x05);
 
-   //Softtakeover for Pitch fader
-   engine.softTakeover("[Channel1]", "rate", true);
-   engine.softTakeover("[Channel2]", "rate", true);
-   engine.softTakeoverIgnoreNextValue("[Channel1]", "rate");
-   engine.softTakeoverIgnoreNextValue("[Channel2]", "rate");
-
-   // Connect the VUMeters
-    engine.connectControl("[Channel1]", "vu_meter", "DJCi300.vuMeterUpdateDeck");
-	engine.getValue("[Channel1]", "vu_meter", "DJCi300.vuMeterUpdateDeck");
-    engine.connectControl("[Channel2]", "vu_meter", "DJCi300.vuMeterUpdateDeck");
-	engine.getValue("[Channel2]", "vu_meter", "DJCi300.vuMeterUpdateDeck");
-    engine.connectControl("[Main]", "vu_meter_left", "DJCi300.vuMeterUpdateMaster");
-    engine.connectControl("[Main]", "vu_meter_right", "DJCi300.vuMeterUpdateMaster");
-	engine.getValue("[Main]", "vu_meter_left", "DJCi300.vuMeterUpdateMaster");
-    engine.getValue("[Main]", "vu_meter_right", "DJCi300.vuMeterUpdateMaster");
+    // Turn On Toneplay LED
+    midi.sendShortMsg(0x96, 0x40, 0x7F);
+    midi.sendShortMsg(0x96, 0x48, 0x7F);
+    midi.sendShortMsg(0x97, 0x40, 0x7F);
+    midi.sendShortMsg(0x97, 0x48, 0x7F);
+
+    // Connect master VUMeters
+    engine.makeConnection("[Master]", "vu_meter_left", DJCi300.vuMeterUpdateMain);
```

Comment:
another [Master]->[Main] update needed I think?

Reply:
Maybe? I was following the wiki, but I guess the wiki might be outdated. It still says to use `VuMeter`, `VuMeterL`, and `VuMeterR`, and Paulus told me those throw deprecated warnings in Zulip.

![image](https://github.com/user-attachments/assets/12e2cc36-6c5e-4104-937c-8eca2e5d0b82)

This is the 2.4 wiki.

Reply:
Many of these changes unfortunately have not made it to the manual page yet. Please try using `[Main]` whenever possible and only revert to `[Master]` when you get errors or broken functionality.